src/pilot/dell_nfv.py

```diff
@@ -0,0 +1,376 @@
+                str(dell_compute_count) +
+                '|" ' +
+                file_path)
+
+            if enable_hugepage:
+                hugepage_number = ConfigOvercloud.calculate_hugepage_count(
+                    hugepage_size)
+                cmds.append('sed -i "s|dellnfv::hugepages::hugepagesize:.*'
+                            '|dellnfv::hugepages::hugepagesize: ' +
+                            hugepage_size[0:-1] + '|" ' + file_path)
+                cmds.append(
+                    'sed -i "s|dellnfv::hugepages::hugepagecount:.*'
+                    '|dellnfv::hugepages::hugepagecount: ' +
+                    str(hugepage_number) +
+                    '|" ' +
+                    file_path)
+            if enable_numa:
+                vcpu_pin_set = ConfigOvercloud.calculate_hostos_cpus(
+                    hostos_cpu_count)
+                cmds.append(
+                    "sed -i 's|dellnfv::numa::vcpu_pin_set:.*"
+                    "|dellnfv::numa::vcpu_pin_set: \"" +
+                    vcpu_pin_set +
+                    "\"|' " +
+                    file_path)
+                cmds.append(
+                    'sed -i "s|  # NovaVcpuPinSet|  ' +
+                    'NovaVcpuPinSet|" ' +
+                    file_path)
+                cmds.append(
+                    "sed -i 's|NovaVcpuPinSet:.*|NovaVcpuPinSet: \"" +
+                    vcpu_pin_set +
+                    "\"|' " +
+                    file_path)
+
+            for cmd in cmds:
+                status = os.system(cmd)
+                logger.debug("cmd: {}".format(cmd))
+            if status != 0:
+                raise Exception(
+                    "Failed to execute the command {}"
+                    " with error code {}".format(
+                        cmd, status))
+
+        except Exception as error:
+            message = "Exception {}: {}".format(
+                type(error).__name__, str(error))
+            logger.error("{}".format(message))
+            raise Exception(
+                "Failed to modify the dell_environment.yaml"
+                " at location {}".format(file_path))
+
+    def get_dell_compute_nodes_hostnames(self, nova):
+        try:
+            logger.info("Getting dellnfv compute node hostnames")
+            # Create host object
+            host_obj = hosts.HostManager(nova)
+
+            # Get list of dell nfv nodes
+            dell_hosts = []
+
+            for host in host_obj.list():
+                if "dell-compute" in host.host_name:
+                    hostname = str(host.host_name)
+                    dell_hosts.append(hostname)
+            return dell_hosts
+        except Exception as error:
+            message = "Exception {}: {}".format(
+                type(error).__name__, str(error))
+            logger.error("{}".format(message))
+            raise Exception("Failed to get the Dell Compute nodes.")
+
+    def edit_aggregate_environment_file(
```

Comment:
I can't find any place where this method is called.  Is it needed?

Reply:
Fixed